src/mgmt/cli_mgmt.c

```diff
@@ -417,6 +428,65 @@ get_attach_info(const char *name, int *npsrs, struct dc_mgmt_psr **psrs,
 	return rc;
 }
 
+/*
+ * Get the CaRT network configuration for this client node
+ * via the get_attach_info() dRPC.
+ * Configure the client's local environment with these parameters
+ */
+int dc_network_cfg(const char *name)
+{
+	int rc;
+	int npsrs;
+	char buf[SYS_INFO_BUF_SIZE];
+	struct sys_info sy_info;
+	struct dc_mgmt_psr *psrs;
+
+	if (name == NULL)
+		name = DAOS_DEFAULT_SYS_NAME;
+
+	/* Query the agent for the CaRT network configuration parameters */
+	rc = get_attach_info(name, &npsrs, &psrs, &sy_info);
+	if (rc != 0)
+		return rc;
+
+	/* These three are always set */
```

Comment:
Would be good to have some sanity-check of this assertion. Is there any way they can wind up empty at this point? If so, it would be preferable to fail noisily here rather than mysteriously later.

Reply:
The CRT_CTX_SHARE_ADDR and CRT_TIMEOUT are integers.  In the worst case, they are 0.  The CRT_PHY_ADDR_STR should never be null.  I could sanity check the data by looking for a NULL string here and implicitly trusting the rest of the data structure if this piece is valid.  The 'always set' logic is used, because those three items have to match the daos_server and allowing a client to override them doesn't really make sense unless we want them to be able to configure the client to fail.  The OFI_INTERFACE and OFI_DOMAIN are truly user configurable based upon the client machine, so it's plausible that the user wants to supply those values to do something special.  Therefore, the client overrides are allowed for those two and not the other three.  Of course, if this is no longer the behavior we want, I will adjust it.

Reply:
I've added a check to the get_attach_info() to validate that the response provided non-null provider, interface and domain strings.  It will now fail noisily at the get_attach_info() call if the data is bad.